### PR TITLE
Terra date input focus style

### PR DESCRIPTION
**Pull request by adoroshk**
### Summary
This PR fixes focus style for `terra-date-input` which was overridden by form-input css (blue solid border appears instead of dashed border).

The solution includs following:
1. Moving `tera-date-input` css styles inside `.date-input {}` [here](https://github.com/cerner/terra-framework/blob/4a9e44470474a91ce206702bf5750ae008cd5eee/packages/terra-date-input/src/DateInput.module.scss#L15). Previously they were declared at the same level. After that change wdio started applying `tera-date-input` styles and stopped overriding them with `terra-form-input` css.
2. Css `border: 0` prop led to no border on focus, or blue `terra-form-input` border on wdio snapshots (see screenshot). That was fixed by adding variables for border on focus and focus+error.
Screenshot 2024-04-10 at 9 04 31 AM

### Testing

This change was tested using:

- [x] WDIO
- [x] Jest
- [x] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

##### Focus.
Screenshot 2024-04-10 at 9 07 37 AM
Screenshot 2024-04-10 at 9 07 45 AM

##### Focus + Error
Screenshot 2024-04-10 at 9 08 04 AM
Screenshot 2024-04-10 at 9 08 10 AM

##### Focus.
Screenshot 2024-04-10 at 9 08 27 AM
Screenshot 2024-04-10 at 9 08 37 AM

##### Focus + Error
Screenshot 2024-04-10 at 9 08 42 AM
Screenshot 2024-04-10 at 9 08 51 AM

##### Focus.
Screenshot 2024-04-10 at 9 09 10 AM
Screenshot 2024-04-10 at 9 09 16 AM

### Reviews

In addition to engineering reviews, this PR needs:

- [x] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-10307

### CODE REVIEW

**Comment by adoroshk** [comment]
Need @eawww help to figure out the correct styling for fusion theme focus if error, as current option on prod looks way different from just on focus:

Screenshot 2024-04-10 at 9 25 40 AM
Screenshot 2024-04-10 at 9 25 45 AM

**Comment by eawww** [comment]
Hi @adoroshk 

For the default theme invalid state, since the visual border becomes wider (i think due to a box shadow) I'd recommend increasing the outline offset by 1px to ensure the required 1px gap between the border and focus indicator. Everything else looks good.

Standard focus indicator for fusion theme on this component's inputs appears to be `border: 1px solid #dedfe0;` `box-shadow: 0 0 8px #a6d9f4;`

**Comment by adoroshk** [comment]
> Hi @adoroshk
> 
> For the default theme invalid state, since the visual border becomes wider (i think due to a box shadow) I'd recommend increasing the outline offset by 1px to ensure the required 1px gap between the border and focus indicator. Everything else looks good.
> 
> Standard focus indicator for fusion theme on this component's inputs appears to be `border: 1px solid #dedfe0;` `box-shadow: 0 0 8px #a6d9f4;`

@eawww I have updated error and error + focus the styling, here is how new snapshot look like:

Default and Lowlight themes Error + Focus outline offset increased by 1px:

Screenshot 2024-04-12 at 4 08 04 PM

Screenshot 2024-04-12 at 4 08 18 PM

Fusion focus styling changed to be `border: 1px solid #dedfe0;` `box-shadow: 0 0 8px #a6d9f4;`:

Screenshot 2024-04-12 at 4 10 13 PM

Screenshot 2024-04-12 at 4 10 20 PM

**I have a question about Fusion Error + focus styling. Currently is is the same as a regular focus, should it be different?**

Screenshot 2024-04-12 at 4 10 28 PM

**Review by BenBoersma — APPROVED** [review]

File: packages/terra-date-input/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Fixed
+  * Focus style fixed to not be overridden by form-input focus styling.
```
Comment (sdadn) [inline packages/terra-date-input/CHANGELOG.md:6]: nit:
```suggestion
  * Fixed focus style to not be overridden by form-input focus styling.
```

**Review by sdadn — APPROVED** [review]

**Pull request merged**